core/src/main/scala/com/lightbend/paradox/ErrorContext.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2015 - 2019 Lightbend, Inc. <http://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lightbend.paradox
+
+import java.io.File
+
+import com.lightbend.paradox.markdown.Page
+import org.pegdown.ast.Node
+
+import scala.collection.mutable.ListBuffer
+
+trait ErrorContext {
+  def apply(msg: String, index: Int): Unit
+
+  def apply(msg: String, node: Node): Unit = apply(msg, node.getStartIndex)
+
+  def apply(msg: String): Unit
+
+  def apply(msg: String, file: File, index: Int): Unit
+
+  def apply(msg: String, file: File, node: Node): Unit = apply(msg, file, node.getStartIndex)
+
+  def apply(msg: String, file: File): Unit
+
+  def apply(msg: String, page: Page, index: Int): Unit = apply(msg, page.file, index)
+
+  def apply(msg: String, page: Page, node: Node): Unit = apply(msg, page.file, node.getStartIndex)
+
+  def apply(msg: String, page: Page): Unit = apply(msg, page.file)
+}
+
+class ErrorCollector extends ErrorContext {
+  private val errors = new ListBuffer[ParadoxError]()
+
+  private def addError(msg: String, page: Option[File], index: Option[Int]) = {
+    errors.append(ParadoxError(msg, page, index))
```

Comment:
While I agree not printing the whole stack trace by default makes sense, I wonder if we should still capture it and print it at `debug` level so it's still available if you need it?

Reply:
I guess if we catch an exception, then yes, but in places where we know exactly where the error from, I don't think it's so important, it's not hard to grep the code for the error messages to see where the message was logged from.

Reply:
So I've updated so everywhere where we swallow exceptions, they are now logged at debug. This involved introducing a new logger abstraction (because so far all logging has just been passing around ad-hoc `String => Unit` functions that come from the sbt logger), so that may be useful in future.